Providers/Modules/NPM/Plugin/plugin/npmd_config_lib.rb

```diff
@@ -629,62 +657,57 @@ def self.getRuleHashFromJson(text)
-                        _rule["NetworkThresholdLoss"] = _test["NetworkThreshold"].nil? ? "-3" : (_test["NetworkThreshold"].has_key?("Loss") ? _test["NetworkThreshold"]["Loss"] : "-2")
-                        _rule["NetworkThresholdLatency"] = _test["NetworkThreshold"].nil? ? "-3.0" : (_test["NetworkThreshold"].has_key?("Latency") ? _test["NetworkThreshold"]["Latency"] : "-2.0")
-                        _rule["ValidStatusCodeRanges"] = _test.has_key?("ValidStatusCodeRanges") ? _test["ValidStatusCodeRanges"] : String.new
-
-                        _connectionMonitorId = _test.has_key?("ConnectionMonitorId") ? _test["ConnectionMonitorId"].to_s : String.new
-
-                        # Iterate over ConnectionMonitorInfoMap to get following info
-                        if !_connectionMonitorId.empty?
-                            _cmMap = _h.has_key?(EpmCMInfoTag) ? _h[EpmCMInfoTag] : Hash.new
-                            if !_cmMap.empty?
-                                _cmId = _cmMap[_connectionMonitorId.to_s]
-                                _rule["CMResourceId"] = _cmId["resourceId"]
-                                _rule["IngestionWorkspaceId"] = _cmId["ingestionWorkspaceId"]
-                                _rule["WorkspaceAlias"] = _cmId["workspaceAlias"]
-                            end
+                _epmRules = {"Rules" => []}
+                _testIds = _h[EpmTestInfoTag]
+                _testIds.each_key do |testId|
+                    _test = _h[EpmTestInfoTag][testId]
+                    _rule = Hash.new
+                    _rule["ID"] = testId
+                    _rule["Name"] = _test["Name"]
+                    _rule["Poll"] = _test["Poll"]
+                    _rule["WorkspaceResourceID"] = @metadata.has_key?("WorkspaceResourceID") ? @metadata["WorkspaceResourceID"] : String.new
+                    _rule["DiscoverPaths"] = _test.has_key?("DiscoverPaths") ? _test["DiscoverPaths"].to_s : "true"
+                    _rule["AppThresholdLoss"] = _test["AppThreshold"].nil? ? "-3" : (_test["AppThreshold"].has_key?("Loss") ? _test["AppThreshold"]["Loss"] : "-2")
+                    _rule["AppThresholdLatency"] = _test["AppThreshold"].nil? ? "-3.0" : (_test["AppThreshold"].has_key?("Latency") ? _test["AppThreshold"]["Latency"] : "-2.0")
+                    _rule["NetworkThresholdLoss"] = _test["NetworkThreshold"].nil? ? "-3" : (_test["NetworkThreshold"].has_key?("Loss") ? _test["NetworkThreshold"]["Loss"] : "-2")
+                    _rule["NetworkThresholdLatency"] = _test["NetworkThreshold"].nil? ? "-3.0" : (_test["NetworkThreshold"].has_key?("Latency") ? _test["NetworkThreshold"]["Latency"] : "-2.0")
+                    _rule["ValidStatusCodeRanges"] = _test.has_key?("ValidStatusCodeRanges") ? _test["ValidStatusCodeRanges"] : nil
+                    _rule["SourceAgentId"] = _test.has_key?("SourceAgentId") ? _test["SourceAgentId"] : nil
```

Comment:
SourceAgentId will not be part of config received from ruby plugin. It has to be filled from AgentIdToTestIdsMap.
On second thought what we should do is while iterating through all the tests, create a vector of sourceAgentIds that this test is part of, and send that vector. I will correct this in Adv Knowledge too.

Reply:
Added new logic for SourceAgentId. It will contain list of agents involved per test.